libs/s25main/desktops/dskCredits.cpp

```diff
@@ -358,7 +359,7 @@ void dskCredits::DrawCredit()
 template<typename T>
 T randEnum()
 {
-    return T(rand() % (helpers::NumEnumValues_v<T> - 2));
+    return T(rand() % (helpers::NumEnumValues_v<T> - 5));
```

Comment:
This looks odd (did before with the `-2`). Where does this magic number come from? Why does it apply to all random enum values generated? It is e.g. used for `Job` but then wine & leather is checked.for in line 404

Reply:
Refactored the drawing code to be more clear and readable.